alm/src/components/ManualExecutionButton.tsx

```diff
@@ -132,15 +141,35 @@ export const ManualExecutionButton = ({
       messageData,
       signatureCollected,
       setExecutionData,
-      setPendingExecution
+      setPendingExecution,
+      safeExecutionAvailable,
+      allowFailures
     ]
   )
 
   return (
-    <div className="is-center">
-      <StyledButton className="button outline" onClick={() => setManualExecution(true)}>
-        Execute
-      </StyledButton>
+    <div>
+      <div className="is-center">
+        <ActionButton className="button outline" onClick={() => setManualExecution(true)}>
+          Execute
+        </ActionButton>
+      </div>
+      {safeExecutionAvailable && (
+        <div
+          title="Allow executed message to fail and record its failure on-chain without reverting the whole transaction.
+          Use fixed gas limit for execution."
+          className="is-center"
+          style={{ paddingTop: 10 }}
+        >
+          <input
+            type="checkbox"
+            id="allow-failures"
+            checked={allowFailures}
+            onChange={e => setAllowFailures(e.target.checked)}
+          />
+          <label htmlFor="allow-failures">Allow Failures</label>
```

Comment:
Does it make sense to call it "Unsafe execution" as so only users which understand what it means consider to use it?

Reply:
Ok, we might do so. I have renamed it to "Unsafe mode".